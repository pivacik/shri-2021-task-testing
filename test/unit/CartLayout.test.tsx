/**
 * @jest-environment jsdom
 */
import React from "react";

import {
  render,
  screen,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { Catalog } from "../../src/client/pages/Catalog";
import { initStore } from "../../src/client/store";
import { CartApi, ExampleApi } from "../../src/client/api";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { CartItem, ProductShortInfo } from "../../src/common/types";
import { AxiosResponse } from "axios";
import { Cart } from "../../src/client/pages/Cart";

function mockResponse<T>(data: T): AxiosResponse<T, any> {
  return {
    status: 200,
    statusText: "OK",
    headers: {},
    config: {},
    data,
  };
}
describe("1", () => {
  it("В каталоге должны отображаться товары, список которых приходит с сервера", async () => {
    const api = new ExampleApi("");
    const product1: CartItem = {
      name: "Pants",
      price: 1102,
      count: 3,
    };
    const product2: CartItem = {
      name: "Chair",
      price: 200,
      count: 1,
    };

    const cart = new CartApi();
    cart.setState({ 1: product1, 2: product2 });
    const store = initStore(api, cart);
    const component = (
      <MemoryRouter>
        <Provider store={store}>
          <Cart />
        </Provider>
      </MemoryRouter>
    );
    const { getByRole } = render(component);
    const table = getByRole("table");
    expect(table).toMatchSnapshot();
  });
});
