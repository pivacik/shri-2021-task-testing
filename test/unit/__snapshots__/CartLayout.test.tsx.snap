// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 В каталоге должны отображаться товары, список которых приходит с сервера 1`] = `
<table
  class="Cart-Table table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        #
      </th>
      <th
        scope="col"
      >
        Product
      </th>
      <th
        scope="col"
      >
        Price
      </th>
      <th
        scope="col"
      >
        Count
      </th>
      <th
        scope="col"
      >
        Total
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      data-testid="1"
    >
      <th
        class="Cart-Index"
        scope="row"
      >
        1
      </th>
      <td
        class="Cart-Name"
      >
        Pants
      </td>
      <td
        class="Cart-Price"
      >
        $
        1102
      </td>
      <td
        class="Cart-Count"
      >
        3
      </td>
      <td
        class="Cart-Total"
      >
        $
        3306
      </td>
    </tr>
    <tr
      data-testid="2"
    >
      <th
        class="Cart-Index"
        scope="row"
      >
        2
      </th>
      <td
        class="Cart-Name"
      >
        Chair
      </td>
      <td
        class="Cart-Price"
      >
        $
        200
      </td>
      <td
        class="Cart-Count"
      >
        1
      </td>
      <td
        class="Cart-Total"
      >
        $
        200
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        colspan="4"
      >
        Order price:
      </td>
      <td
        class="Cart-OrderPrice"
      >
        $
        3506
      </td>
    </tr>
  </tfoot>
</table>
`;
